### ISSUE TITLE
errors in log and daily sensor 

### ISSUE THREAD

**Issue by ciechompl** [jeroenterheerdt/HAsmartirrigation#23]
upgraded to the latest version (0.0.18), removed intergation, re-added from scratch and got errors in log 
```
2020-05-28 19:37:54 ERROR (MainThread) [custom_components.smart_irrigation.sensor] argument of type 'float' is not iterable
2020-05-28 19:37:54 ERROR (MainThread) [custom_components.smart_irrigation.sensor] argument of type 'float' is not iterable
2020-05-28 19:37:54 ERROR (MainThread) [custom_components.smart_irrigation.sensor] argument of type 'float' is not iterable
2020-05-28 19:37:54 ERROR (MainThread) [custom_components.smart_irrigation.sensor] argument of type 'float' is not iterable
2020-05-28 19:38:24 ERROR (MainThread) [custom_components.smart_irrigation.sensor] argument of type 'float' is not iterable
2020-05-28 19:38:24 ERROR (MainThread) [custom_components.smart_irrigation.sensor] argument of type 'float' is not iterable
2020-05-28 19:38:24 ERROR (MainThread) [custom_components.smart_irrigation.sensor] argument of type 'float' is not iterable
2020-05-28 19:38:24 ERROR (MainThread) [custom_components.smart_irrigation.sensor] argument of type 'float' is not iterable
```
+ when running reset.bucket or calculate daily service my sensor goes into state unavailable for a few minutes ... 
![image](https://user-images.githubusercontent.com/27088154/83175035-95584380-a11b-11ea-8eae-376e73db8256.png)

![image](https://user-images.githubusercontent.com/27088154/83175049-9b4e2480-a11b-11ea-99b5-bb3d7534854e.png)

after few minutes calculates some strange/high values: 
![image](https://user-images.githubusercontent.com/27088154/83175574-61315280-a11c-11ea-97ff-f1250933546a.png)

![image](https://user-images.githubusercontent.com/27088154/83175596-6a222400-a11c-11ea-8b06-2e1df6c80a70.png)

**Comment by jeroenterheerdt** [comment]
yeah, I noticed a bug after 40 minutes myself, should be fixed in the release I just pushed: https://github.com/jeroenterheerdt/HAsmartirrigation/releases/tag/v0.0.19

**Comment by ciechompl** [comment]
hehe upgraded .but now services look differently :) did you rename them ? or just simply need to re-add integration ? 

![image](https://user-images.githubusercontent.com/27088154/83178879-7a88cd80-a121-11ea-91fb-b714c31dc588.png)

**Comment by jeroenterheerdt** [comment]
yeah, this was done as part of the #21 . Also note #22, so expect further automation name changes.

**Comment by jeroenterheerdt** [comment]
can you confirm the issue has been solved so we can close this issue?

**Comment by ciechompl** [comment]
nope ... no errors in log but still sensor get "unavailable" state 

re services ... i missed info about renaming but is it ok that you allow whitespace in service name ?  it works now but not sure whether this is in line with the general naming convention in HA.

**Comment by jeroenterheerdt** [comment]
yeah, that is a bug, working on it now.

**Comment by jeroenterheerdt** [comment]
check: https://github.com/jeroenterheerdt/HAsmartirrigation/releases/tag/v0.0.20

**Comment by ciechompl** [comment]
you are so fast in reacting on reported issues that I can even follow everything :) thx, that's really cool  :)  

re sensor state still goes to "unavailable" whenever i run reset bucket or daily service 
the only difference is that when i use reset bucked it goes to unavailable for couple of minutes only ... after daily service seems state stays unavailable longer but finally it is set properly (= hourly time)  so I think this is ok acceptable now. we can close it I believe.

**Comment by jeroenterheerdt** [comment]
Please let me know if this still exists in future releases. I think removing, rebooting, installing should bring you back to a better state. This is not the way it is supposed to work.

**Comment by ciechompl** [comment]
ok will test it again with next release .. I've doing this order: upgrade, reboot, remove, re-add .. so maybe I did it in wrong order .. anyway now it is acceptable as after few minutes I'm getting proper states

**Comment by jeroenterheerdt** [comment]
please uninstall, reboot, reinstall and see if the issue is gone. you should get states right away, not after a while.

**Comment by posixx** [comment]
After removing integrations, restarting hass and addin integrations again all is wel.. The only issues logged are a couple of warnings:

`2020-05-28 22:02:47 WARNING (MainThread) [custom_components.smart_irrigation.config_flow] config_entry: <homeassistant.config_entries.ConfigEntry object at 0x7f7a540135f0>
2020-05-28 22:03:00 WARNING (MainThread) [custom_components.smart_irrigation.config_flow] update_options
2020-05-28 22:03:00 WARNING (MainThread) [custom_components.smart_irrigation] async_reload_entry
2020-05-28 22:03:00 WARNING (MainThread) [custom_components.smart_irrigation] async unload entry
2020-05-28 22:03:00 ERROR (MainThread) [custom_components.smart_irrigation.sensor] 'float' object has no attribute 'split'
2020-05-28 22:03:00 ERROR (MainThread) [custom_components.smart_irrigation.sensor] 'float' object has no attribute 'split'
2020-05-28 22:03:00 ERROR (MainThread) [custom_components.smart_irrigation.sensor] 'float' object has no attribute 'split'
2020-05-28 22:03:00 ERROR (MainThread) [custom_components.smart_irrigation.sensor] 'float' object has no attribute 'split'
2020-05-28 22:05:00 WARNING (MainThread) [custom_components.smart_irrigation.config_flow] config_entry: <homeassistant.config_entries.ConfigEntry object at 0x7f7a5404ceb0>
2020-05-28 22:05:04 WARNING (MainThread) [custom_components.smart_irrigation.config_flow] update_options
2020-05-28 22:05:04 WARNING (MainThread) [custom_components.smart_irrigation] async_reload_entry
2020-05-28 22:05:04 WARNING (MainThread) [custom_components.smart_irrigation] async unload entry
2020-05-28 22:05:04 ERROR (MainThread) [custom_components.smart_irrigation.sensor] 'float' object has no attribute 'split'
2020-05-28 22:05:04 ERROR (MainThread) [custom_components.smart_irrigation.sensor] 'float' object has no attribute 'split'
2020-05-28 22:05:04 ERROR (MainThread) [custom_components.smart_irrigation.sensor] 'float' object has no attribute 'split'
2020-05-28 22:05:04 ERROR (MainThread) [custom_components.smart_irrigation.sensor] 'float' object has no attribute 'split'`

But as this is only during first setup i wouldn't bother..

**Comment by ciechompl** [comment]
ok i check it again ... sensor state is being updated as you said ... immediately ... but as I've been displaying result in minutes in my frontend (not the state in seconds) i haven't notices where the error really is .. so ...whenever I run service (reset bucket or daily) all additional attributes are gone for few minutes: 

![image](https://user-images.githubusercontent.com/27088154/83191642-a6fa1500-a134-11ea-9dbe-ef35397d7c10.png)

that is why I got unavailable as attribute adjusted_run_time_minutes is missing .. .so loot at this part of the code not the sensor state update. 
normal vie at the sensor: 
![image](https://user-images.githubusercontent.com/27088154/83192318-a2822c00-a135-11ea-9bef-ea1e21392c3b.png)

second observation is that hourly service seems not to update hourly sensor ... I did my value to hourly sensor e.g. 1200s (via perl script) but when started smart_irrigation.smart_irrigation_calculate_hourly_adjusted_run_time nothing has changed in the hourly sensor state... it returned back to its normal value after few minutes.  
attributes haven't disappeared as in case of daily sensor: 

![image](https://user-images.githubusercontent.com/27088154/83192207-7ff01300-a135-11ea-9544-51f91060db98.png)
